src/generate-query.ts

```diff
@@ -508,10 +505,56 @@ function getMissingSlicingArg(
   )
   if (usesSlicingArg) return null
 
-  // Return the first slicing arguments:
-  return field.arguments.find((arg) =>
-    slicingArguments.includes(arg.name.value)
+  // Returns the first slicing argument
+  // if slicing argument can be split on '.'
+  // then we have a nested slicing argument,
+  // and we need to follow the path all the way
+  // down to the leaf input field definition
+  return field.arguments.find((arg) => {
+    return slicingArguments.find((slicingArgument) => {
```

Comment:
Can you remind me why we need to follow all the way down to the leaf input field definition when we are just returning the top level argument? I feel like there's something strange about this whole thing or I'm just not understanding it correctly.

Reply:
We need to follow to make sure that path down actually exists. I guess, what you are saying is that this repo is not about actually verifying that the path exists, but about returning the top level argument that matches?

Reply:
Yeah, is this library responsible for checking if the slicing argument is valid or if the list size directive is valid as a whole? I'm not sure.

Reply:
I agree, it is not this repo's responsibility. I will remove this, make it a single level match.

Reply:
resolving convo, let me know if you disagree with how I went about it and have more suggestions